plugins/out_s3/s3.c

```diff
@@ -83,6 +83,27 @@ int s3_plugin_under_test()
     return FLB_FALSE;
 }
 
+struct flb_aws_header *create_canned_acl_header(char *canned_acl)
+{
+    struct flb_aws_header *h = NULL;
+    
+    if (canned_acl == NULL) {
+        return NULL;
+    }
+    h = flb_malloc(sizeof(struct flb_aws_header));
+    if (h == NULL) {
+        flb_errno();
+        return NULL;
+    } 
+   
+    h->key = "x-amz-acl";
+    h->key_len = 9;
+    h->val = canned_acl;
+    h->val_len = flb_sds_len(canned_acl);
```

Comment:
canned_acl is a `char *` based on the function declaration. flb_sds_len can only be called on a `flb_sds_t`- which is an alias for char *, though that doesn't mean SDS strings are just char*. You can read this here to understand how SDS strings work, there's some cool memory tricks involved: https://github.com/antirez/sds

Anyway, in this function either you should use strlen() or you must require that the function argument is an SDS

Reply:
Thanks so much